### DISCUSSION TITLE
3D feature extractor

### DISCUSSION THREAD

**Discussion by joaolcguerreiro** [Project-MONAI/MONAI#6621] [Q&A]
Hi!

I just found out that MONAI has [pre-trained 3d models](https://docs.monai.io/en/stable/networks.html#nets). I was wondering if any of them can be used for feature extraction in order to compute a perceptual loss...

Is this possible with any of the provided networks? If it is possible do we need to normalize the input with some dataset statistics?

Thank you :)

**Comment by KumoLiu** [comment]
Hi @joaolcguerreiro, IMO, CNN, RNN, and Transformer can all be considered feature extractors for extracting data features.
Thanks!

**Reply by joaolcguerreiro** [reply]
I am looking for pre-trained models that handle 3D data. MONAI nets don't seem to support this right now. Do you know any alternative?

**Reply by Wangrui-berry** [reply]
model = monai.networks.nets.resnet10(spatial_dims=3, n_input_channels=1, n_classes=1)
What's the meaning? Can it solve your problem?

**Reply by Wangrui-berry** [reply]
https://docs.monai.io/en/stable/_modules/monai/networks/nets/resnet.html
    conv_type: Callable = Conv[Conv.CONV, spatial_dims]
    conv_type: type[nn.Conv1d | nn.Conv2d | nn.Conv3d] = Conv[Conv.CONV, spatial_dims]
    It seems could solve your problem. I am also trying to use 3d network to process 3d data using Monai.

**Reply by KumoLiu** [reply]
Hi @joaolcguerreiro, since most pre-trained weights are based on the ImageNet which is a 2d dataset.
https://github.com/Project-MONAI/MONAI/blob/a89a0af10e70256fbe8641e71385ad2c2b3fb405/monai/networks/blocks/fcn.py#L112
You can try to use the pre-trained weights introduced in https://monai.io/model-zoo.html.

Thanks!
